src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs

```diff
@@ -65,16 +65,31 @@ private static NetworkCredential GetCredentialsFromString(string value)
             {
                 return null;
             }
-            int idx = value.IndexOf(":");
-            if (idx < 0)
+
+            // The User and password may or may not be URL encoded.
+            // Curl seems to accept both. To match that we do opportunistic decode.
+            try
             {
-                // only user name without password
-                return new NetworkCredential(value, "");
+                value = Uri.UnescapeDataString(value);
```

Comment:
>Uri.UnescapeDataString(value); [](start = 24, length = 30)

This seems incorrect: if my password has some fragment that looks like an escape sequence but I wanted it literally it will be converted to something else, no?

Reply:
Or is it guaranteed that at this point it was previously escaped?

---
In reply to: [174844816](https://github.com/dotnet/corefx/pull/28088#discussion_r174844816) [](ancestors = 174844816)

Reply:
If it is guaranteed could you please change the comment on top and the parameter name to make that clear.

---
In reply to: [174847585](https://github.com/dotnet/corefx/pull/28088#discussion_r174847585) [](ancestors = 174847585,174844816)